web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java

```diff
@@ -125,6 +136,41 @@ public String getViewablePdfFilePid(ContentObjectRecord briefObj, AccessGroupSet
         return null;
     }
 
+    /**
+     * Get the first content object that has an original file datastream
+     * and return it if the user has the appropriate permissions
+     * @param briefObj
+     * @param principals
+     * @return
+     */
+    public ContentObjectRecord getContentObject(ContentObjectRecord briefObj, AccessGroupSet principals) {
+        if (permissionsHelper.hasOriginalAccess(principals, briefObj)) {
+            return briefObj;
+        }
+
+        ContentObjectRecord contentObj = getChildFileObject(briefObj, principals);
```

Comment:
I'm realizing that getChildFileObject should return null immediately if the briefObj it is passed is anything other than a work, otherwise this call will attempt to find the children of file objects. FileObjects don't have queryable children according to the solr index, they just have datastreams.